packages/react-dom-bindings/src/events/plugins/FormActionEventPlugin.js

```diff
@@ -81,18 +86,16 @@ function extractEvents(
       // It should be in the document order in the form.
       // Since the FormData constructor invokes the formdata event it also
       // needs to be available before that happens so after construction it's too
-      // late. The easiest way to do this is to switch the form field to hidden,
-      // which is always included, and then back again. This does means that this
-      // is observable from the formdata event though.
-      // TODO: This tricky doesn't work on button elements. Consider inserting
-      // a fake node instead for that case.
+      // late. We use a temporary fake node for the duration of this event.
       // TODO: FormData takes a second argument that it's the submitter but this
       // is fairly new so not all browsers support it yet. Switch to that technique
       // when available.
-      const type = submitter.type;
-      submitter.type = 'hidden';
+      const temp = submitter.ownerDocument.createElement('input');
```

Comment:
think adding `.type = 'hidden'` to the temp node is any more efficient?

Reply:
You mean because it won't invalidate layout or something? Not sure that's true since even hidden forms can be made to participate in layout.

Reply:
It's also invalid on button so wouldn't make any difference in most cases.